### PR TITLE
feat: ui for autocommit

### LINKED ISSUE #29127: [Task]: UI changes for auto-commit
### Is there an existing issue for this?

- [X] I have searched the existing issues

### SubTasks

- Switch in admin settings
- Bottom bar changes

### PR DESCRIPTION
**Pull request by brayn003**
## Description
- Adds a button to enable/disable auto-commit in Git Settings
- Adds loading state for auto-commit

#### PR fixes following issue(s)
Fixes #29127 

#### Media
image

- New feature (non-breaking change which adds functionality)
>
>
>
- [ ] Manual
- [ ] JUnit
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

## Summary by CodeRabbit

- **New Features**
  - Introduced auto-commit functionality for Git synchronization.
  - Added modals and status bars to manage and display auto-commit progress.
  - Implemented feature flags to enable or disable the auto-commit feature.

- **Enhancements**
  - Updated Git settings to include auto-commit configuration options.
  - Improved user interface with new messages and warnings related to auto-commit.

- **Bug Fixes**
  - Addressed issues with Git-related actions to ensure smoother user experience.

- **Documentation**
  - Added user-facing messages and descriptions for the auto-commit feature.

- **Refactor**
  - Refactored Git settings components to accommodate new auto-commit feature.

- **Style**
  - Updated styles to support new components related to auto-commit functionality.

### CODE REVIEW

File: app/client/src/pages/Editor/gitSync/QuickGitActions/AutocommitStatusbar.tsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  completed,
+  onHide,
+}: AutocommitStatusbarProps) {
+  const period = 4;
+  const [percentage, setPercentage] = useState(0);
+  useEffect(() => {
+    if (completed) {
+      setPercentage(100);
+      if (onHide) {
+        setTimeout(() => {
```
Comment (sneha122) [inline app/client/src/pages/Editor/gitSync/QuickGitActions/AutocommitStatusbar.tsx:42]: Can we please store this setTimeout instance in the variable and clear it when we no longer require it?

File: app/client/src/pages/Editor/gitSync/Tabs/GitSettings/index.tsx [hunk 2]
```diff
@@ -32,7 +32,7 @@ function GitSettings() {
             <GitProtectedBranches />
           </>
         ) : null}
-        {isGitAdmin && <GitDisconnect />}
+        {isGitAdmin && <DangerZone />}
```
Comment (sneha122) [inline app/client/src/pages/Editor/gitSync/Tabs/GitSettings/index.tsx:35]: Curious, is there any reason why we changed from GitDisconnect to DangerZone?

Comment (brayn003) [reply app/client/src/pages/Editor/gitSync/Tabs/GitSettings/index.tsx:35]: Yes, because DangerZone now contains GitDisconnect and Disable AutoCommit

**Review by sneha122 — APPROVED** [review]

**Review by Aishwarya-U-R — APPROVED** [review]
Proxy approval.

**Pull request merged**